tests/asteroid_agent_test.py

```diff
@@ -23,17 +25,37 @@ def testAsteroidAgent_whenExploitCheckDetectVulnz_EmitsVulnerabilityReport(
 
 
 def testAsteroidAgent_whenTooManyRedirects_doesNotCrash(
-    exploit_instance_with_report: Iterator[Type[definitions.Exploit]],
     asteroid_agent_instance: asteroid_agent.AsteroidAgent,
     agent_mock: list[m.Message],
+    mocker: plugin.MockerFixture,
+    requests_mock: requests_mock.Mocker,
 ) -> None:
     """Ensure that the agent does not crash when there are too many redirects."""
+
+    def response_callback(request: Any, context: Any) -> str:
```

Comment:
Where are the "Too many redirects" implemented in this test?

Reply:
![image](https://github.com/user-attachments/assets/710ae446-5418-45cd-a063-48414c94f362)
This is how requests procs redirects
```python
        context.headers = {"Location": request.url}
        context.status_code = 302
        return ""
```

Reply:
The header location, is taking it to the same url over and over and over until whatever max redirect we have is exceeded